src/ConnectedMode/Migration/FileProviders/XmlFileProvider.cs

```diff
@@ -39,16 +39,19 @@ namespace SonarLint.VisualStudio.ConnectedMode.Migration.FileProviders
     internal class XmlFileProvider : IFileProvider
     {
         private readonly IServiceProvider serviceProvider;
+        private readonly IRoslynProjectProvider projectProvider;
         private readonly ILogger logger;
         private readonly IThreadHandling threadHandling;
         private readonly IFileSystem fileSystem;
 
+        // We only search for non-project files - we get the Roslyn project file paths
+        // from the VSWorkspace.
+        // Corner case: this means that we won't clean projects that are in the solution
+        // but unloaded, since they won't appear in the VsWorkspace
```

Comment:
That way, there's no risk that we pick up any projects that are not part of the solution.

Also, it means we can skip looking for other types of file to clean up: no Roslyn projects = no files to check for settings.